Makefile

```diff
@@ -19,11 +20,21 @@ cli:
 	$(MAKE) --environment-overrides -C internal/cmd/gencli cli
 
 .PHONY: tools
-tools:
+tools: golangci-lint
 	go generate -tags tools tools/tools.go
 	go install github.com/bufbuild/buf/cmd/buf@v1.3.1
 	go install github.com/mfridman/tparse@v0.10.3
 
+# golangci-lint recommends installing the binary directly, instead of using go get
+# See the note: https://golangci-lint.run/usage/install/#install-from-source
+.PHONY: golangci-lint
+golangci-lint:
+	$(eval GOLINT_INSTALLED := $(shell which golangci-lint))
+
+	if [ "$(GOLINT_INSTALLED)" = "" ]; then \
+		curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GO_PATH)/bin v1.50.1; \
```

Comment:
Is this really the best way of doing this? I'm a bit worried about introducing something like this in our CI pipeline. Instead, could we copy the contents of this (or at least the important parts) to a local script that we control, instead of one that could be modified in a supply chain attack?